src/client/app/needs/needs.html

```diff
@@ -1,20 +1,33 @@
-<section>
-    <div class="contentWrapper container">
-        <div class="row">
-            <category-search class="col-xs-12 col-sm-4 col-md-4 col-lg-4 marginBottom" category="vm.category"></category-search>
-            <location-search class="col-xs-12 col-sm-4 col-md-4 col-lg-4 marginBottom" location="vm.location"></location-search>
-            <div class="col-xs-12 col-sm-4 col-md-4 col-lg-4 marginBottom">
-                <div class="input-group">
-                    <input type="text" class="form-control" placeholder="Я шукаю..." ng-model="vm.searchValue" ng-keyup="$event.keyCode === 13 && vm.setSearch()">
-                    <span class="input-group-btn">
-                        <button class="btn btn-default" type="button" ng-click="vm.setSearch()">Go!</button>
-                    </span>
+<section id="needs">
+    <section class="main-filters">
+        <div class="container">
+
+            <div class="col-xs-12 col-sm-11 no-gutter">
+                <div class="col-xs-6 no-gutter margin-bottom-20">
+                    <input id="key-search" type="search" ng-model="vm.preSearch"
+                           ng-keyup="$event.keyCode === 13 && vm.submitCategorySearch(vm.preSearch, vm.category, vm.location)"
```

Comment:
Same thing regarding "vm.submitCategorySearch", check line 8.
You are overwriting working code with old one.